src/include/sof/audio/component.h

```diff
@@ -425,6 +425,15 @@ struct comp_driver_info {
 	struct list_item list;		/**< list of component drivers */
 };
 
+/**
+ * \struct comp dev memory - used like devm_kzalloc
+ * \brief comp dev memory block - used for every memory allocated by comp
+ */
+struct comp_dev_mem {
+	void *ptr; /**< A pointr to particular memory block */
+	struct list_item mem_list; /**< list of memory allocated by codec */
```

Comment:
@lgirdwood would you have an issue if I built in a feature (behind a config flag) here to track amount allocated by each alloc? That way we call a function and query memory consumed by a component (and possibly identify what section each pointer is point to)

Reply:
@cujomalainey sound good to me.